src/shim.js

```diff
@@ -2,9 +2,69 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+/* global browser */
+
 "use strict";
 
+const providerEvents = [
```

Comment:
Please call this `PROVIDER_EVENTS` to keep with convention. I'd actually prefer `PROVIDER_EVENT_NAMES`. Could you also alphabetize it please?